Feature/itbl 3347 universal link redirects

davidtruong opened the pull request:
Added a URLSession delegate which handles the dataTaskWithURL task. Stops the request after a response and doesn't make the redirect call. Returns the redirected url to the callback of the getAndTrackRequest call.

davidtruong committed:
Working version of the no-redirect call to dataTaskWithURL.

Need to verify the effects on our existing api calls if those hit a re-directed link. Perhaps we should create a different session for the getAndTrack call which specifies the re-direct logic.

davidtruong committed:
Removes the extra reference to the session.

davidtruong committed:
Updated version which abstracts out the NSURLSessionDelegate from IterableApi for deep linking

davidtruong committed:
fixed incorrect static function reference

davidtruong committed:
updated names for the urlSession to be redirectUrlSession not to be confused with the iterableApi session

davidtruong committed:
fix name redirectUrlSession

davidtruong commented:
@dontgitit Any thoughts on if I should leave getAndTrackDeeplink as a instance function vs making it static? It can be an instance function because getAndTrackRedirect gets called in continueUserActivity after IterableApi is created in application willFinishLaunchingWithOptions.

Originally I made it an instance so that it could share the same session. However it turns out that the request should be on a different session so it doesn't break our other api calls if they happen to make a redirect.

dontgitit commented on Iterable-iOS-SDK/IterableDeeplink.m:
```diff
@@ -0,0 +1,74 @@
+static NSURLSession *redirectUrlSession = nil;
+NSString *deepLinkLocation = nil;
+
+// documented in IterableAPI.h
```
IterableDeeplink.h

dontgitit reviewed (approved):
lftm assuming it works

dontgitit commented:
@davidtruong whatever makes the most sense/is easiest to use for people implementing it

davidtruong committed:
Refactor of IterableDeeplink into IterableDeeplinkManager
Changed getAndTrackDeeplink back to be static since it doesn't require an instance of IterableApi

davidtruong committed:
updating documentation and minor changes to static fields

davidtruong commented:
Refactored the class to be called IterableDeeplinkManager and use a singleton.
Changed getAndTrackDeeplink back to be static so that is backwards compatible with applications that already have the sdk integrated.
